components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/bindings/handlers/TokenBindingExpiryEventHandler.java

```diff
@@ -100,6 +113,124 @@ public void handleEvent(Event event) throws IdentityEventException {
         }
     }
 
+    /**
+     * This method will get the application information from session context and revoke access tokens of the
+     * applications bound to that session. This method can be used when token binding information id not found in the
+     * request.
+     *
+     * @param event Event.
+     * @throws IdentityOAuth2Exception
+     * @throws InvalidOAuthClientException
+     */
+    private void revokeAccessTokensForSSOSessionBinding(Event event)
+            throws IdentityOAuth2Exception, InvalidOAuthClientException {
+
+        String sessionContextIdentifier = getSessionIdentifier(event);
+        Map<String, Object> eventProperties = event.getEventProperties();
+        if (StringUtils.isNotBlank(sessionContextIdentifier)) {
+            SessionContext sessionContext = (SessionContext) eventProperties.get(IdentityEventConstants
+                    .EventProperty.SESSION_CONTEXT);
+            if (sessionContext != null) {
+                Map<String, SequenceConfig> authenticatedSequences = sessionContext.getAuthenticatedSequences();
+                if (MapUtils.isEmpty(authenticatedSequences)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Could not find the authenticated sequence config map in the session context");
+                    }
+                    return;
+                }
+
+                AuthenticatedUser user = (AuthenticatedUser) sessionContext
+                        .getProperty(FrameworkConstants.AUTHENTICATED_USER);
+                if (user == null) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Authenticated user is nnot founnd in the session context");
+                    }
+                    return;
+                }
+                String tenantDomain = user.getTenantDomain();
+                if (StringUtils.isBlank(tenantDomain)) {
+                    if (log.isDebugEnabled()) {
+                        log.debug("Could not find the tenant domain of the user: " + user.getUserName());
+                    }
+                    return;
+                }
+                for (Map.Entry<String, SequenceConfig> sequenceConfigEntry : authenticatedSequences.entrySet()) {
+                    String applicationName = sequenceConfigEntry.getValue().getApplicationId();
+                    String clientId = getClientIdFromApplicationName(applicationName, tenantDomain);
+                    String bindingType = OAuth2Util.getAppInformationByClientId(clientId).getTokenBindingType();
+                    if (OAuth2Constants.TokenBinderType.SSO_SESSION_BASED_TOKEN_BINDER.equals(bindingType)) {
```

Comment:
Shall we add a debug log if the binding is not SSO binding, for given session identitifer?